### DISCUSSION TITLE
Howto run a groovy file from netbeans 21?

### DISCUSSION THREAD

**Discussion by patton73** [apache/netbeans#7120] [General]
I cannot run a groovy script since the "run" is always disabled. I can only compile it.

**Comment by matthiasblaesing** [comment]
Show your project, show sample code (minimal project, zipper), maybe then people can help you and don't need to guess what you try to do.

**Comment by patton73** [comment]
Minimal project zip? Are you joking? Just try to run a groovy script in a gradle java project in NB 21. Simply you will not find a way.
Left click on the file and click "Run" was too hard to implement?

**Comment by geertjanw** [comment]
Be polite and kind to those who are working 100% free for you, @patton73. Provide what is asked for and you will be helped.

**Comment by matthiasblaesing** [comment]
@patton73 fine, I'll bite and do your work:

Steps (this was done on Ubuntu Linux):

0. Start a terminal and enter a folder you can cleanup and that has no folder with name `netbeans` and `testuserdir` in it
1. `wget https://dlcdn.apache.org/netbeans/netbeans/21/netbeans-21-bin.zip`
2. `unzip netbeans-21-bin.zip`
3. `./netbeans/bin/netbeans --userdir testuserdir`
4. Do not import old settings (!)
5. Go to `Tools` -> `Plugins` -> `Installed`, Check `Groovy` and click on `Activate`, wait till finished
6. Close Pluginmanager
7. Open Favorites (Window -> Favorites)
8. Go to temp folder (for example the folder you used to download NB21 to
9. Open `New file` dialog: Right click on folder entry, choose `New`, choose `Other`
10. Select category `Other`, Choose `Empty File`, Choose Next, Enter `test.groovy`, Enter `println "Hello World!"`
11. Right click into editor window, in the Menu you find "Run File" there: 
![image](https://github.com/apache/netbeans/assets/2179736/8ae632d9-9e49-49d2-ac02-016724b3dd5b)
12. Running works:
![image](https://github.com/apache/netbeans/assets/2179736/a0a10359-5264-4631-a79a-388bdc05d183)

On the file entry you find a similar menu.
![image](https://github.com/apache/netbeans/assets/2179736/7350d6b2-dae7-4fae-8223-b0c64c74823a)

**Comment by patton73** [comment]
Thanks a lot for the complete tutorial. Anyway to me the "run file" in the various menues is never active.  And groovy plugin is installed and activated.
But now that i know that it should be possible to run a groovy file in this way i will check my NB configuration for any possible misconfiguration. 
Thanks again.

**Comment by halfwind22** [comment]
@matthiasblaesing Since this thread isn't closed yet, just wanted to ask something: The steps mentioned by you work perfectly, but is it the only way forward to work with Groovy projects in Netbeans. For comparison, Eclipse has this concept of "Groovy projects" that gives an easy to use method of working with Groovy scripts and later executing them, especially when there are lots of Groovy files with packages and all that.  But here, looks like I will have to navigate to the  directory myself and create the folder structure(for packages) , instead of the IDE helping me out.
